index.js

```diff
@@ -34,16 +34,32 @@ module.exports = (input, columns, options) => {
 	}
 
 	if (position === 'start') {
-		return ellipsis + sliceAnsi(input, length - columns + 1, length);
+		if (space === true) {
+			ellipsis += ' ';
+		}
+
+		return ellipsis + sliceAnsi(input, length - columns + (space ? 2 : 1), length);
 	}
 
 	if (position === 'middle') {
+		if (space === true) {
+			ellipsis = ' ' + ellipsis + ' ';
+		}
+
 		const half = Math.floor(columns / 2);
-		return sliceAnsi(input, 0, half) + ellipsis + sliceAnsi(input, length - (columns - half) + 1, length);
+		return (
+			sliceAnsi(input, 0, half) +
+			ellipsis +
+			sliceAnsi(input, length - (columns - half) + (space ? 3 : 1), length)
```

Comment:
Too many magic numbers. `3` and `1` should be variables with a descriptive name. Same in the other places.

Reply:
I named it `ellipsisWidth`. :)